modules/loader/src/main/java/com/opengamma/strata/loader/CharSources.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.loader;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.ByteSource;
+import com.google.common.io.CharSource;
+import com.google.common.io.Files;
+import com.opengamma.strata.collect.io.ResourceLocator;
+
+/**
+ * Helper that allows {@link CharSource} objects to be created.
+ */
+public final class CharSources {
+
+  /**
+   * Obtains an instance of {@link CharSource} from a file name, specified as a String.
+   *
+   * @param fileName the file name, as a String
+   * @return a new instance of {@link CharSource} with the default charset on the current JVM.
+   */
+  public static CharSource ofFileName(String fileName) {
+    return Files.asCharSource(new File(fileName), Charsets.UTF_8);
+  }
+
+  /**
+   * Obtains an instance of {@link CharSource} from a file name, specified as a String. This also takes in a specific character set, as a {@link Charset}.
```

Comment:
Each of these CharSet Javadoc lines is too long. The extra sentence should be on a new line.